internal/manifest/oci_index.go

```diff
@@ -102,13 +118,23 @@ func (index *OCI1IndexPublic) editInstances(editInstances []ListEdit) error {
 				return fmt.Errorf("update %d of %d passed to OCI1Index.UpdateInstances had no media type (was %q)", i+1, len(editInstances), index.Manifests[i].MediaType)
 			}
 			index.Manifests[targetIndex].MediaType = editInstance.UpdateMediaType
+			if editInstance.UpdateAffectAnnotations {
+				index.Manifests[targetIndex].Annotations = maps.Clone(editInstance.UpdateAnnotations)
+			} else {
+				for k, v := range editInstance.UpdateAnnotations {
+					index.Manifests[targetIndex].Annotations[k] = v
+				}
```

Comment:
`maps.Copy(index.Manifests[targetIndex].Annotations, editInstance.UpdateAnnotations)`

Reply:
While working with tests i found that `Copy` panics if  `dest` is `nil`. So we must set `nil` to `map[string]string{}` before doing copy.